code/modules/power/lighting.dm

```diff
@@ -290,62 +295,86 @@
 	return
 
 /**
-  * Updates the light's properties
+  * Updates the light's 'on' state and power consumption based on [/obj/machinery/light/var/on].
   *
-  * Updates the icon_state, luminosity, colour, and power usage of the light.
-  * Also handles rigged light bulbs exploding.
   * Arguments:
-  * * trigger - Should this update make the light explode/burn out? (Defaults to TRUE)
+  * * trigger - Should this update check if the light will explode/burn out.
+  * * instant - Will the lightbulb turn on instantly, or after a short delay.
+  * * play_sound - Will the lightbulb play a sound when it's turned on.
   */
-/obj/machinery/light/proc/update(trigger = TRUE)
+/obj/machinery/light/proc/update(trigger = TRUE, instant = FALSE, play_sound = TRUE)
 	switch(status)
 		if(LIGHT_BROKEN, LIGHT_BURNED, LIGHT_EMPTY)
 			on = FALSE
-	update_icon()
-	if(on)
-		var/BR = brightness_range
-		var/PO = brightness_power
-		var/CO = brightness_color
-		if(color)
-			CO = color
-		var/area/A = get_area(src)
-		if(A && A.fire)
-			CO = bulb_emergency_colour
-		else if(nightshift_enabled)
-			BR = nightshift_light_range
-			PO = nightshift_light_power
-			if(!color)
-				CO = nightshift_light_color
-		var/matching = light && BR == light.light_range && PO == light.light_power && CO == light.light_color
-		if(!matching)
-			switchcount++
-			if(rigged)
-				if(status == LIGHT_OK && trigger)
-					log_admin("LOG: Rigged light explosion, last touched by [fingerprintslast]")
-					message_admins("LOG: Rigged light explosion, last touched by [fingerprintslast]")
-					explode()
-			// Whichever number is smallest gets set as the prob
-			// Each spook adds a 0.5% to 1% chance of burnout
-			else if(prob(min(40, switchcount / 10)))
-				if(status == LIGHT_OK && trigger)
-					burnout()
-
-			else
-				use_power = ACTIVE_POWER_USE
-				set_light(BR, PO, CO)
-	else
+	if(on) // Turning on
+		if(instant)
+			_turn_on(trigger, play_sound)
+		else if(!turning_on)
+			turning_on = TRUE
+			addtimer(CALLBACK(src, .proc/_turn_on, trigger, play_sound), rand(LIGHT_ON_DELAY_LOWER, LIGHT_ON_DELAY_UPPER))
+	else // Turning off
 		use_power = IDLE_POWER_USE
 		set_light(0)
+		update_icon()
 
 	active_power_usage = (brightness_range * 10)
 	if(on != power_state) // Light was turned on/off, so update the power usage
 		power_state = on
 		if(on)
-			static_power_used = brightness_range * 20 //20W per unit luminosity
+			static_power_used = brightness_range * 20 //20W per unit of luminosity
 			addStaticPower(static_power_used, STATIC_LIGHT)
 		else
 			removeStaticPower(static_power_used, STATIC_LIGHT)
 
+/**
+  * The actual proc to turn on the lightbulb.
+  *
+  * Do not call directly, use [/obj/machinery/light/proc/update] instead.
```

Comment:
You may be able to use one of the spacemandmm helpers here to ensure you cant invoke this like a public method
```dm
	SHOULD_BE_PURE(TRUE)
	PRIVATE_PROC(TRUE)
	PROTECTED_PROC(TRUE)
```

One of these will work, see which one trips the vscode DM builder up
![image](https://user-images.githubusercontent.com/25063394/130069865-e84525ae-c936-44d7-ae88-3e095d729de6.png)

You can click this to test without doing the entire DM rebuild

Reply:
Oh nice, I was wondering if it had something like that. Couldn't find any documentation for the spacemandmm stuff though so I just left it.